PatientWebApplication/HealthClinic.DAL/Model/Patient/DoctorAppointment.cs

```diff
@@ -28,5 +28,18 @@ public DoctorAppointment(int id, TimeSpan time, string date, int patientId, int
 
         public DoctorAppointment() : base() { }
 
+        public override bool Equals(Object obj)
+        {
+            var item = obj as DoctorAppointment;
+            if (item == null)
+            {
+                return false;
+            }
+            else
+            {
+                return this.Date.Equals(item.Date) && this.DoctorUserId == item.DoctorUserId && this.PatientUserId == item.PatientUserId && this.RoomId.Equals(item.RoomId);
+            }
```

Comment:
Else statement unnecessary.

Reply:
Use the method to check the equality of two strings, to check the date as well as to check the RoomId.